Lib/subprocess.py

```diff
@@ -1096,9 +1096,18 @@ def _execute_child(self, args, executable, preexec_fn, close_fds,
 
             assert not pass_fds, "pass_fds not supported on Windows."
 
-            if not isinstance(args, str):
+            if isinstance(args, str):
+                pass
+            elif not shell and isinstance(args, os.PathLike):
+                args = list2cmdline([os.fsdecode(args)])
+            else:
+                args = list(args)
+                args[0] = os.fsdecode(args[0])
                 args = list2cmdline(args)
 
+            if executable is not None:
+                executable = os.fsdecode(executable)
+
```

Comment:
I still think bytes and pathlike should be supported for every item of `args`. The POSIX implementation doesn't limit this to `args[0]` (though not documented), and I think that's proper. A script shouldn't have to special-case calling `os.fsdecode` for a pathlike argument. That's inefficient for POSIX, for which `os.fsencode` would be better. While `os.fspath` would be efficient, it's insufficient because it might return bytes on Windows.

Reply:
The POSIX implementation accepts also bytes as `args` and as any item of `args`. If allow path-like arguments, we should consider also adding support of bytes arguments and program name.